service/worker/scheduler/workflow.go

```diff
@@ -249,7 +249,14 @@ func (s *scheduler) run() error {
 	s.pendingPatch = s.InitialPatch
 	s.InitialPatch = nil
 
-	for iters := s.tweakables.IterationsBeforeContinueAsNew; iters > 0 || s.pendingUpdate != nil || s.pendingPatch != nil; iters-- {
+	iters := s.tweakables.IterationsBeforeContinueAsNew
+	for {
+		// TODO: use the real GetContinueAsNewSuggested
+		suggestContinueAsNew := iters <= 0 || workflow.GetInfo(s.ctx).GetCurrentHistoryLength() >= 1e6
```

Comment:
should/could `1e6` be a named constant? until I read the test, I didn't really understand why it was there and that it wouldn't have any effect in production.